### DISCUSSION TITLE
Strange Problem with one device

### DISCUSSION THREAD

**Discussion by userMak** [krahabb/meross_lan#290] [General]
Hi
I have several mss520h 2.0.0 switches 
Only one of them is constantly like disconnected. It is blinking red for some seconds and then turns green. Even when it is blinking red I can turn it off/on.

in the logs I get the following
![image](https://github.com/krahabb/meross_lan/assets/55662933/0ac68498-a55a-4d91-9b99-ed88c9656142)

in the devices section I have the following
in the logbook for example it says turned off several times although we haven't used the switch at the time.

![image](https://github.com/krahabb/meross_lan/assets/55662933/3f221ad2-887d-4a22-93ad-8e38118aa85f)

I don't understand what it means the following
"incorrect timestamp: 1685260018 seconds behind HA"

I have reset it several times but the problem remains
Is there something I can do to fix this?

**Comment by krahabb** [comment]
Hello @userMak,
the log warning actually tells us the device is unable to get the correct epoch from time servers (NTP) and this usually occurs at device boot (and periodically after that, some users did some tracking/snifffing on this)
The blinking is usually the MQTT broker connection procedure: i.e. if blinking red it's not connected to any MQTT broker
I don't remember if you're using your devices binded to a private broker or to the Meross app but in both scenarios the red blinking has the same meaning
As for the repeating states this is maybe due to a very quick disconnection detected by meross_lan but not logged in history since the disconnection happens for too short in the polling cycle and is immediately overwrited by the new state reading (which is the same as before the disconnection)
This could mean the device is effectively rebooting every so often (and the red blinking is due to MQTT reconnection after boot)

One thing that comes to mind (I also have a FritzBOX) is that the router is maybe limiting the device access to the internet?

**Reply by userMak** [reply]
Just check the fritzbox. Evrything looks fine there..
